client/pagination.go

```diff
@@ -0,0 +1,60 @@
+package client
+
+import "strconv"
+
+const limit = 100
+
+type Pagination struct {
+	offset int
+	params map[string]string
+}
+
+type Paginated interface {
+	Environment
+	getEndpoint() string
+}
+
+func (p *Pagination) getParams() map[string]string {
+	return map[string]string{
+		"limit":  strconv.Itoa(limit),
+		"offset": strconv.Itoa(p.offset),
+	}
+}
+
+// returns true if there is more data.
+func (p *Pagination) next(currentPageSize int) bool {
+	p.offset += currentPageSize
+	return currentPageSize == limit
+}
+
+// params - additional params. may be nil.
+func getAll[P Paginated](client *ApiClient, params map[string]string) ([]P, error) {
+	p := Pagination{
+		offset: 0,
+		params: params,
+	}
+
+	var allResults []P
+
+	for {
+		pageParams := p.getParams()
+		for k, v := range params {
```

Comment:
merge the offset, limit params with the other params (may be nil if there are no other params).